File: ui/components/ui/ProducerSession/message-generator/data-generators/json-generator/fixed-json-generator.ts
import * as pb from '../../../../../../grpc-web/tools/teal/pulsar/ui/producer/v1/producer_pb';

export type FixedJsonGenerator = {
  type: 'fixed-json-generator',
  json: string
};

export function fixedJsonGeneratorFromPb(v: pb.FixedJsonGenerator): FixedJsonGenerator {
  return {
    type: 'fixed-json-generator',
    json: v.getJson()
  };
}

export function fixedJsonGeneratorToPb(v: FixedJsonGenerator): pb.FixedJsonGenerator {
  const message = new pb.FixedJsonGenerator();
  message.setJson(v.json);

  return message;
}

File: server/src/main/scala/producer/message_generator/data_generators/json_generator/FixedJsonGenerator.scala
package producer.message_generator.data_generators.json_generator

import com.tools.teal.pulsar.ui.producer.v1.producer as pb

case class FixedJsonGenerator(
    json: String
):
    def generate: String = json

object FixedJsonGenerator:
    def fromPb(v: pb.FixedJsonGenerator): FixedJsonGenerator =
        FixedJsonGenerator(
            json = v.json
        )

    def toPb(v: FixedJsonGenerator): pb.FixedJsonGenerator =
        pb.FixedJsonGenerator(
            json = v.json
        )
